Add Query action BBE (#3231)
ballerina-platform/ballerina-distribution: examples/query-actions/query_actions.md

> # Query Actions

Reviewer: ```suggestion
# Query actions
```

> The clauses in a query-action are executed in the same way as the clauses in a query-expr.

Reviewer: ```suggestion
The clauses in a query-action are executed in the same way as the clauses in a query-expression.
```

Reviewer 2: ```suggestion
The clauses in a query action are executed in the same way as the clauses in a query expression.
```

> In the same way the query is built in query expressions, the query can be built using other clauses such as the `let` clause, `join` clause, `order by` clause, `where clause`, and `limit` clause except the `select` clause.

Reviewer 3: ```suggestion
Query action starts with the `from` clause and ends with the `do` clause.
In the same way, the query is built in query expressions, query actions also can have intermediate clauses of  `let`, `join`, `order by`, `where`, and `limit` clauses.
```

> In the same way the query-pipeline is built in query expressions, the query pipeline can be built using other clauses such as `let` clause, `join` clause, `order by` clause, `where clause` and `limit` clause except the `select` clause.

Reviewer 2: Our query expression-related BBEs don't mention about the `query pipeline`? Wouldn't it be confusing for a new user, to what a query pipeline is since we have not explained it? Maybe we can word this differently.

> If a clause in the query action completes early with error `e`, the result of the query-action is `e`. Otherwise, the result of the query-action is `nil`.

Reviewer: ```suggestion
If a clause in the query action completes early with an error `e`, the result of the query-action is `e`. Otherwise, the result of the query-action is `nil`.
```

Reviewer 2: ```suggestion
If a clause in the query action completes early with an error `e`, the result of the query action is `e`. Otherwise, the result of the query action is `nil`.
```

# Query actions

The clauses in a query action are executed in the same way as the clauses in a query expression.

Query action starts with the `from` clause and ends with the `do` clause.

> In the same way, the query is built in query expressions, query actions also can have intermediate clauses of  `let`, `join`, `order by`, `where`, and `limit` clauses.

Reviewer 3: ```suggestion
In the same way, the query is built in query expressions, query actions also can have intermediate clauses of  `let`, `join`, `order by`, `where`, and `limit`.
``
